### PR TITLE
LPS-96933 | Standardize ajax requests in frontend-js-aui-web

### PR DESCRIPTION
**Pull request by markocikos**
Hi @jbalsas  ,

Couple of notes:
- `restore_entry.js` I was unable to test: unused, deprecated
- `inline_editor_base.js` I was unable to test. I don't think we are using `Liferay.CKEditorInline` anywhere, `inlineEditSaveURL` is not set anywhere.
- `store.js`: I didn't replace `io.request`, as it will be removed as a part of https://issues.liferay.com/browse/LPS-95863

Please review the code.

Thank you!

_This is a redirection from https://github.com/wincent/liferay-portal/pull/45_

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js [hunk 1]
```diff
@@ -489,19 +489,19 @@ AUI.add(
 
 				A.mix(data, options);
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/layout.js]: Would be interesting to track things like these so we can eventually remove the dependency on `aui-base` and `Liferay.provide` in the future

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/logo_editor.js [hunk 2]
```diff
@@ -81,13 +81,12 @@ AUI.add(
...
 					var instance = this;
 
-					instance.publish({
-						uploadComplete: {
-							defaultFn: A.rbind('_defUploadCompleteFn', instance)
-						},
-						uploadStart: {
-							defaultFn: A.rbind('_defUploadStartFn', instance)
-						}
+					instance.publish('uploadComplete', {
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/logo_editor.js]: Why this change? Are functions properly bound to the `instance` in all cases after this?

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js [hunk 3]
```diff
@@ -166,19 +166,20 @@ AUI.add(
...
-					});
+					var data = {
+						className: instance.get('className'),
+						classPK: instance.get('classPK'),
+						p_auth: Liferay.authToken,
+						p_l_id: themeDisplay.getPlid(),
+						score: score
+					};
+
+					Liferay.Util.fetch(url, {
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/ratings.js]: /cc @adolfopa, @boton, @ambrinchaudhary, are you still using this `ratings` component? Could you double-check this?

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/upload.js [hunk 4]
```diff
@@ -937,28 +937,28 @@ AUI.add(
...
-										li
-									);
-								}
-							}
-						});
+						var data = {
+							fileName: li.attr('data-fileName')
+						};
+
+						Liferay.Util.fetch(deleteFile, {
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/upload.js]: Hey @adolfopa, @boton, @ambrinchaudhary... seems this component is heavily used in your apps. Could you please do a quick double check before we push it forward?

**Comment by jbalsas** [comment]
ci:test

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
Just started reviewing :)

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/logo_editor.js (hunk 2 above)
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/logo_editor.js]: @jbalsas The change was to reduce over-engineering, as a part of slightly different arguments propagation. But, it is true, we cannot be 100% sure it's not needed, specially if some customer solution is relying on this event. I reverted the change, pushed as a new commit.

**Comment by jbalsas** [comment]
Hey @john-co, could you analyze this PR? We'd like to merge and backport, so I'd like to make sure it's safe.

**Comment by ambrinchaudhary** [comment]
Hey @markocikos, @jbalsas : 

although the changes in `ratings.js` and `upload.js` looks good to me, I have run `ant all` with the code and found some errors in portal. It would be great if someone could take a look.  

> In portal, go to **Content&Data > Knowledge Base** or **Documents and Media** => Control Menu renders nothing but a "\\" character, and there are errors both in server and browser. 

By the way, I have tested ratings in Blog's widget and works fine!

**Comment by markocikos** [comment]
@ambrinchaudhary @jbalsas 
> In portal, go to Content&Data > Knowledge Base or Documents and Media => Control Menu renders nothing but a "\" character, and there are errors both in server and browser.

I was able to reproduce this on a clean `master`, the issue is unrelated to this PR. There are a bunch of unsatisfied OSGi references in the log, and NPEs when accessing page.

**Comment by ambrinchaudhary** [comment]
Hey @markocikos, 

I am sorry. Yesterday I checked in master and I didn't get that error, but today I am seeing it, so yes, must be unrelated to your PR... apologises.

**Comment by markocikos** [comment]
@ambrinchaudhary Dropping database and making new one fixed the issue for me.

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test

**Comment by jbalsas** [comment]
let's see how this looks one last time.... @john-co, could you please review test results?

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test

**Comment by jbalsas** [comment]
I'm still seeing a lot of failures here to feel comfortable. @john-co, could you please analyze the results once the get back?

**Comment by jbalsas** [comment]
ci:test

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
I couldn't replicate three unique failures from https://github.com/jbalsas/liferay-portal/pull/1825#issuecomment-516478567

**Comment by john-co** [comment]
ci:test:relevant

**Comment by john-co** [comment]
ci:test

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Hi @jbalsas, the failures look unrelated to the pull :heavy_check_mark: 

cc/ @markocikos

**Comment by jbalsas** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/76565. [See changes here.](https://github.com/jbalsas/liferay-portal/compare/1abf2e6f4ab5~1...1abf2e6f4ab5) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jbalsas** [comment]
Hey @markocikos, I've forwarded this to Brian. A couple of notes:
- We need to remove the dependencies we clean up from the [module definitions in modules.js](https://github.com/brianchandotcom/liferay-portal/pull/76565/commits/1abf2e6f4ab5d56ee4dbc22919734b4fb32b4e63#diff-1dfd821f4fdf0a612b24b05dd9e731f7) or specific `config.js` files as well as from the module implementation.
- After this, I still see some usages of `io-request` in the module:
    - I assume we're getting rid of this for `Store` as part of https://github.com/jbalsas/liferay-portal/pull/1835
    - Inside [`language.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/language.js#L48) through `A.io`
    - Inside [`poller.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/poller.js#L164) through `A.io`

Could you see if any of those need any additional work done?

Thanks!

**Comment by jbalsas** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/76567. [See changes here.](https://github.com/jbalsas/liferay-portal/compare/4a3c8ee5f005~1...4a3c8ee5f005) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by markocikos** [comment]
@jbalsas 
> * After this, I still see some usages of `io-request` in the module:
>   
>   * I assume we're getting rid of this for `Store` as part of #1835

Yes, there would have been conflicts if I removed it here.

>   * Inside [`language.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/language.js#L48) through `A.io`
>   * Inside [`poller.js`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/poller.js#L164) through `A.io`
> 
> Could you see if any of those need any additional work done?

Ooh, I did not know of this syntax, searched for `io.request`. I opened a new task for this, https://issues.liferay.com/browse/LPS-99339. These are the only usages of this syntax.

Do you know any other syntax that I might be missing?

**Comment by jbalsas** [comment]
> Do you know any other syntax that I might be missing?

Not off the top of my head
